### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by illwieckz**
Many improvements, original work discussed in #1, commits from [mkeditorpacks](https://github.com/illwieckz/mkeditorpacks) were merged back in this tree (can be split later as submodule if needed).

- update unvanquished website urls in menus
- update q3map2 build profile (inspired from Ingar's own gamepack)
- better source reorganisation
- tools are in a dedicated directory (that could be a submodule in the future)
- build is makefile driven
- updated READMEs

notable change: build files are now in a `build/` parent dir. This means both netradiant and gtkradiant build script must use this new path to get their gamepack. By the way, no one supports unvanquished and no one fetches the unvanquished gamepack yet, so it's just changing something that is not yet used.

_[Edit: see [illwieckz/unvanquished-mapeditor-support/tree/improvements](https://github.com/illwieckz/unvanquished-mapeditor-support/tree/improvements) for a preview]_

### CODE REVIEW

File: README.md
Comment (mjstahlberg) [inline README.md:13]: Could also keep the readme files in `src` or in a `doc` folder and copy them into `build` for consistency.

File: README.md
Comment (mjstahlberg) [inline README.md]: See comment above about the readme files.

File: build/gtkradiant/README.md
Comment (mjstahlberg) [inline build/gtkradiant/README.md:15]: I have just uploaded my backup of this repo at https://github.com/Viech/GtkRadiant, just so you know.

File: build/netradiant/README.md
Comment (mjstahlberg) [inline build/netradiant/README.md:1]: Could also have moved this file instead of creating a new one with the same content. 😉 

Also, better phrasing wouldn't hurt here, though that goes for a lot of the english text in this repository…

File: build/netradiant/install/pkg/default_shaderlist.txt
Comment (mjstahlberg) [inline build/netradiant/install/pkg/default_shaderlist.txt]: Again, why pre-populate a build directory? Potentially other editors may need the same file, too.

File: mkeditorpacks/README.md
Comment (mjstahlberg) [inline mkeditorpacks/README.md]: Should rather add a proper license file and/or license headers. Ideally we should contact neumond about it as he started working on the scripts. Right now they are effectively copyrighted by him if I'm not mistaken.

File: mkeditorpacks/mkeditorpacks
Comment (mjstahlberg) [inline mkeditorpacks/mkeditorpacks]: Your classic input sanitizer that might as well be a backdoor itself!

File: mkeditorpacks/tools/buildmenu.py
Comment (mjstahlberg) [inline mkeditorpacks/tools/buildmenu.py]: Thanks!

**Review by mjstahlberg — COMMENTED** [review]
Had a quick look at the diffs. Do you want me to also test generation?

File: mkeditorpacks/README.md
Comment (neumond) [reply mkeditorpacks/README.md]: @Viech you can change the license to any suitable permissive or copyleft license.

File: mkeditorpacks/mkeditorpacks
Comment (illwieckz) [reply mkeditorpacks/mkeditorpacks]: I plan to rewrite it in python

File: build/gtkradiant/README.md
Comment (illwieckz) [reply build/gtkradiant/README.md:15]: I already pulled that commit from older pull request (GitHub keeps pull request data even when repository/branches are deleted), this is a rebased branch with fixes:
https://github.com/illwieckz/GtkRadiant/tree/unvanquished

File: README.md
Comment (illwieckz) [reply README.md:13]: good idea

File: build/netradiant/README.md
Comment (illwieckz) [reply build/netradiant/README.md:1]: […] though that goes for a lot of english text I edit :wink:

File: README.md
Comment (illwieckz) [reply README.md:13]: done

File: mkeditorpacks/mkeditorpacks
Comment (illwieckz) [reply mkeditorpacks/mkeditorpacks]: rewritten to python

**Comment by illwieckz** [comment]
> Do you want me to also test generation?

@Viech why not, I now added a `make clean` target.

File: build/netradiant/install/pkg/default_shaderlist.txt
Comment (illwieckz) [reply build/netradiant/install/pkg/default_shaderlist.txt]: in fact this file was not generated for netradiant in initial @neumond work, I guess it's an error and a gtkradiant specific stuff. By the way this file was not pre-populated, but built (copied), but it's hard to guess since git does not know what a copy is.

**Comment by illwieckz** [comment]
except English phrasing, it looks like I addressed all the issues reported.

**Comment by illwieckz** [comment]
for information, there is no one shell script anymore, everything is now python3 driven.

File: mkeditorpacks/README.md
Comment (illwieckz) [reply mkeditorpacks/README.md]: Hey @neumond! Thanks you for that answer! :smiley: 
@Viech , which license do you prefer? for this kind of scripts a permissive one like BSD or ISC looks good IMO. I personally like the ISC one for its concision, but it looks like the BSD one is already used by Unvanquished/Daemon stuff, do what you prefer!

Comment (neumond) [reply mkeditorpacks/README.md]: Not a problem, @illwieckz, glad to see someone finds this work useful :)

Comment (illwieckz) [reply mkeditorpacks/README.md]: For you information you will find your pk3dir NetRadiant commits there (working branch):
https://gitlab.com/illwieckz/netradiant/commits/pk3dir
and your GtkRadiant ones there:
https://github.com/illwieckz/GtkRadiant/tree/unvanquished
badly, for this one, it builds but does not browse pk3dirs, I haven't figured out why yet. The pk3dir support you wrote in NetRadiant is far more advanced and featureful though.

All that works is very useful, thanks to have done it!

Comment (mjstahlberg) [reply mkeditorpacks/README.md]: Hi @neumond, what's up? 😃

Let's make it BSD 3-clause, as the script could potentially be used by other Dæmon-based games. @neumond, @illwieckz should we use `Unvanquished Development` (and potentially `Daemon Developers` in case we move this to @DaemonEngine at some point) as the copyright holder or do you prefer our names? Copyrighted years are 2014 and 2016–2017, by the way.

**Comment by mjstahlberg** [comment]
I get a different `<key name="template_version" value="6351" />` than the versioned one in `build/gtkradiant/install/pkg/scripts/default_project.proj` but otherwise I can reproduce your build.

Is there maybe a more elegant way of providing the generated files without having them versioned in this repository? Some sort of CI tool with packaged output that we can just link to?

Also, do you think it's worthwhile to move the generation scripts to @DaemonEngine and work with submodules? I wouldn't do that in this very PR, just a general question.

**Comment by illwieckz** [comment]
> I get a different <key name="template_version" value="6351" /> than the versioned one

From `gtkradiant:install/installs/Q3Pack/install/baseq3/scripts/default_project.proj`:
```xml
<!-- bump template_version to force a regen of the user project file from the template. Useful when tweaking the commands -->
<key name="template_version" value="8"/>
```
From `mkeditorpacks/tools/buildmenu.py`:
```python
    d = datetime.utcnow()
    tv = d.day + d.month * 32 + (d.year - 2000) * 367
    add('template_version', tv) # increase every update
```

Yes, it's ugly, that's why I really don't like GtkRadiant.

> Is there maybe a more elegant way of providing the generated files without having them versioned in this repository? Some sort of CI tool with packaged output that we can just link to?

Well, we can make the build/ directory as a submodule and git push it after generation… But it will ends with as many as submodules as fingers in one hand for just a couple files… :cry:

> Also, do you think it's worthwhile to move the generation scripts to @DaemonEngine and work with submodules?

That was my first want, by the way, I merged them back because there is no urge to do it, and I prefer to see the changes merged without having to wait on the answer "do I have to publish that tool on @DaemonEngine and under which name", and well, I just want to get rid of it if possible.

**Comment by mjstahlberg** [comment]
Yeah as I said, there is no urge in splitting and moving the repository. All in all it looks really good to me, I would just fix the license before a merge.

File: src/gamefile/file.game
Comment (illwieckz) [inline src/gamefile/file.game]: If advertized `webp` support is missing in NetRadiant, NetRadiant will crash. Perhaps would it be safer to not advertise this format yet, or makes NetRadiant just raise a warning on unknown formats.

Comment (mjstahlberg) [reply src/gamefile/file.game]: I'm not sure if we use webp anywhere in our asset base but in my humble opinion it has only caused trouble for us and we should not advertise its use at the moment anyway.

File: mkeditorpacks/README.md
Comment (illwieckz) [reply mkeditorpacks/README.md]: Hey, I missed that comment! So what I have to do? add a COPYING file with BSD 3-clause inside and some headers to python files? I'm OK with `Unvanquished Development`, there is names in git history. Is years 2014-2017 OK?

**Comment by illwieckz** [comment]
I added license file and headers in `mkeditorpacks`.

I also removed the `omap.py` file since it was not used.

File: mkeditorpacks/LICENSE.md
Comment (mjstahlberg) [inline mkeditorpacks/LICENSE.md:1]: That year doesn't seem right. Not that we care for this anywhere else in our various codebases, just saying. 😛

File: mkeditorpacks/mkeditorpacks
Comment (mjstahlberg) [inline mkeditorpacks/mkeditorpacks]: Strictly speaking, you may use the dash between years only if every year was actually one where the code was touched by the named copyright holder. Though again, it's not like we cared much before.

**Review by mjstahlberg — APPROVED** [review]
License nitpicks shouldn't prevent a merge. Go ahaed!

File: mkeditorpacks/mkeditorpacks
Comment (illwieckz) [reply mkeditorpacks/mkeditorpacks]: I never understood what the date is meaning on this scope. Copyright expiration is based on time after author death, not based on time after the file was created or last touched.

File: mkeditorpacks/LICENSE.md
Comment (illwieckz) [reply mkeditorpacks/LICENSE.md:1]: in fact scripts in `mkeditorpacks/tools/` by @neumond have history since 2014.

File: mkeditorpacks/mkeditorpacks
Comment (illwieckz) [reply mkeditorpacks/mkeditorpacks]: fixed

**Comment by illwieckz** [comment]
> License nitpicks shouldn't prevent a merge. Go ahaed!

nitpicks fixed too :stuck_out_tongue: 

so let's go!

**Pull request merged**
